### PR TITLE
read topic from file if not exist

### PR DESCRIPTION
**Pull request by MXueguang**

### CODE REVIEW

File: pyserini/search/_base.py [hunk 1]
```diff
@@ -142,6 +142,16 @@ def get_topics(collection_name):
         topics = JTopicReader.getTopicsWithStringIds(JTopics.DPR_SQUAD_TEST)
     elif collection_name == 'dpr-curated-test':
         topics = JTopicReader.getTopicsWithStringIds(JTopics.DPR_CURATED_TEST)
+    elif os.path.exists(collection_name):
```
Comment (lintool) [inline pyserini/search/_base.py:145]: hrm, the issue with this is that we break the qid order... can we do the thing with ms marco where we preserve the iteration order?

**Review by lintool — COMMENTED** [review]
thoughts?

File: pyserini/search/_base.py (hunk 1 above)
Comment (MXueguang) [reply pyserini/search/_base.py:145]: pass a new option, say `args.order` to use predefined query order?
https://github.com/castorini/pyserini/blob/df569681b70b08b15d37ae2e80b088c238cc3353/pyserini/search/__main__.py#L231

another choice maybe require the topic file name contains specific prefix or suffix

Comment (lintool) [reply pyserini/search/_base.py:145]: No need for option... I think it should just be the default impl.

I can't think of an instance where you'd _not_ want this... can you?

Comment (MXueguang) [reply pyserini/search/_base.py:145]: but how do we know which topic the topic file belongs to _by default_?
- msmarco-doc-dev
- msmarco-doc-test
- msmarco-passage-subset-dev
- msmarco-passage-subset-test

Comment (MXueguang) [reply pyserini/search/_base.py:145]: from file name?

Comment (lintool) [reply pyserini/search/_base.py:145]: yea, if the user specifies a file, we should respect the line ordering.

if it's a "known" collection, we do whatever the default is...

**Review by lintool — APPROVED** [review]

**Pull request merged**